streams/src/main/java/org/apache/kafka/streams/kstream/JoinWindows.java

```diff
@@ -31,7 +31,9 @@
  *     WHERE
  *       stream1.key = stream2.key
  *       AND
- *       stream2.ts - before <= stream1.ts <= stream2.ts + after
+ *       stream2.ts >= stream1.ts - before
+ *       AND
+ *       stream2.ts <= stream1.ts + after
```

Comment:
We could, although I note that such an SQL query would likely produce incorrect results.

```sql
stream1.ts - before <= stream2.ts <= stream1.ts + after
```
is likely to be interpreted as
```sql
(stream1.ts - before <= stream2.ts) <= stream1.ts + after
```
If the `stream2` record is too early to fall within the `before` window, you would end up with
```sql
(false) <= stream1.ts + after
```
But since `<=` is a numerical comparison, `false` is likely to be casted into `0`, which results in:
```sql
0 <= stream1.ts + after
```
which will evaluate to `true`, even though the `stream2` record actually falls too early and should not be joined.